research-software-directory/research-software-directory
# Expand data model for project collection

This PR:
- removes the project scraper and its associated crontab job; 
- expands the schema to feed upcoming changes (a template for project pages)
- adds a data migration script to mangle the 2.0.x data into 2.1.x

Refs #409

- project api https://localhost/api/project works
- individual project https://localhost/api/project?title=GlamMap works 
- resolved and cached version https://localhost/api/project_cache works  
- harvester works
- http://localhost/admin shows the admin interface


Steps to test it out:
1. make sure you have no lingering state from work you did previously, see "removing local state" section in the main README

1. 
    ```
    docker-compose build
    source rsd-secrets.env
    docker-compose up -d && docker-compose logs --follow
    ```

1. wait till the RSD is up, start other terminal

    ```
    source rsd-secrets.env
    docker-compose exec database mongo rsd    # (start mongo shell in the database service)
    # copy paste data-migration/data-migration-2.0.x-to-2.1.x.js to update the database data
    ```
1. http://localhost/api/project?title=PuReGoMe to check (or leave the query part out to get all projects)
1. `exit` mongo shell
1. 
    ```
    docker-compose exec harvesting python app.py resolve projects
    ```
1. http://localhost/api/project_cache?title=PuReGoMe to check (or leave the query part out to get all projects)
1. http://localhost/admin to check if that works
